x-pack/plugins/upgrade_assistant/public/application/components/kibana_deprecations/kibana_deprecations.tsx

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState, useCallback } from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+
+import {
+  EuiButtonEmpty,
+  EuiPageBody,
+  EuiPageHeader,
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import type { DomainDeprecationDetails } from 'kibana/public';
+import { SectionLoading } from '../../../shared_imports';
+import { useAppContext } from '../../app_context';
+import { NoDeprecationsPrompt } from '../shared';
+import { KibanaDeprecationList } from './deprecation_list';
+import { StepsModal, StepsModalContent } from './steps_modal';
+import { KibanaDeprecationErrors } from './kibana_deprecation_errors';
+import { ResolveDeprecationModal } from './resolve_deprecation_modal';
+import { LEVEL_MAP } from '../constants';
+
+const i18nTexts = {
+  pageTitle: i18n.translate('xpack.upgradeAssistant.kibanaDeprecations.pageTitle', {
+    defaultMessage: 'Kibana',
+  }),
+  pageDescription: i18n.translate('xpack.upgradeAssistant.kibanaDeprecations.pageDescription', {
+    defaultMessage: 'Some Kibana issues may require your attention. Resolve them before upgrading.',
```

Comment:
Review the issues listed here and make the necessary changes before upgrading.
Critical issues must be resolved before you upgrade.